Automattic/wp-calypso
# Add CalypsoI18nProvider to layout (3)

#### Changes proposed in this Pull Request

This PR creates a `CalypsoI18nProvider` component for Calypso (see #42836) and uses it to wrap all of calypso that uses the layout module. 

The immediate purpose of this is to allow translations to be provided for the `@automattic/composite-checkout` package (part of #42831), but it has wide-ranging uses.

Most of the code in `CalypsoI18nProvider` was copied directly from [gutenboarding](https://github.com/Automattic/wp-calypso/blob/master/client/landing/gutenboarding/index.tsx). 

#### Testing instructions

- Apply the changes from [this other PR](https://github.com/Automattic/wp-calypso/pull/43112) by running `git cherry-pick bb0022588def11551e671bed8d8983e0fe66c21f`
- Visit http://calypso.localhost:3000/me/account and change your language to French.
- Add a plan to your cart and visit checkout. You'll need to add `?flags=composite-checkout-force` to the URL to override the locale detection that blocks it for now.
- Verify that the "Continue" buttons read "Continuer". These strings are localized _inside_ the `composite-checkout` package and not in Calypso.

![Screen Shot 2020-06-04 at 5 33 25 PM](https://user-images.githubusercontent.com/2036909/83813425-4fefc500-a68b-11ea-8356-31143e12ac05.png)
